src/http_api_handler.erl

```diff
@@ -89,21 +89,27 @@ handle_request(<<"POST">>, _, json, Req, State) ->
             {true, Req2, State}
     end;
 
+handle_request(<<"DELETE">>, <<"/api/v1/contexts">>, json, Req, State) ->
+    ok = ergw_api:delete_contexts(length(ergw_api:contexts(all))),
+    Response = jsx:encode(#{contexts => length(ergw_api:contexts(all))}),
+    Req2 = cowboy_req:set_resp_body(Response, Req),
+    {true, Req2, State};
+
 handle_request(<<"DELETE">>, <<"/api/v1/contexts/", _/binary>>, json, Req, State) ->
     Value = cowboy_req:binding(count, Req),
+    ContextsLen = length(ergw_api:contexts(all)),
     case catch binary_to_integer(Value) of
-	Count when is_integer(Count), Count > 0 ->
+	Count when is_integer(Count), Count > 0, ContextsLen >= Count ->
 	    ok = ergw_api:delete_contexts(Count),
-	    Contexts = ergw_api:contexts(all),
-
-	    Response = jsx:encode(#{contexts => erlang:length(Contexts)}),
+	    Response = jsx:encode(#{contexts => length(ergw_api:contexts(all))}),
 	    Req2 = cowboy_req:set_resp_body(Response, Req),
 	    {true, Req2, State};
 	_ ->
```

Comment:
We get to here when Value is not an integer or < 0, and that is fine.
But what about ContextsLen >= Count? Please add a comment why this is not the same as deleting all.